### ISSUE TITLE
Actionlint does not report expressions that reference github "Default environment variables" via f.x `${{ env.CI }}`

### ISSUE THREAD

**Issue by georgespalding** [rhysd/actionlint#264]
I'm still learning the nitty gritty details of github pipelines and made a mistake, which I expected `actionlint` to report.

The following job uses an invalid expression which if I understand the docs can never have a value (but I falsely assumed default env vars could be referenced this way):

```yaml
jobs:
  job1:
    runs-on: ubuntu-latest
    outputs:
      output1: ${{ steps.step1.outputs.test }}
    steps:
      - id: step1
        run: echo "test=hello" >> ${{ env.GITHUB_OUTPUT }}
  job2:
    runs-on: ubuntu-latest
    needs: job1
    steps:
      - run: echo ${{needs.job1.outputs.output1}}
```

The linter gladly accepted this, but pipelines reported an error like:

```text
Run run: echo "test=hello" >> 
/home/runner/work/_temp/bc198ppp-5969-4beb-9aca-fa92b5578c95.sh: line 1: syntax error near unexpected token `newline'
Error: Process completed with exit code 2.
```

Reading about [Naming conventions for environment variables](https://docs.github.com/en/actions/learn-github-actions/variables#naming-conventions-for-environment-variables),
I don't think that any [Default environment variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) are valid when referenced like this: `${{ env.GITHUB_WORKSPACE }}`.

Being new to `actionlint`, and still learning github pipelines, it would have been very helpful to have `actionlint` report that this was an invalid expression.

Would it be possible to extend `actionlint` to check for expressions referencing default variables this way?

**Comment by rhysd** [comment]
This is intended behavior. You should replace

```yaml
run: echo "test=hello" >> ${{ env.GITHUB_OUTPUT }}
```

with

```yaml
run: echo "test=hello" >> $GITHUB_OUTPUT
```

Since it cannot be detected statically that `env.GITHUB_OUTPUT` is empty, actionlint cannot catch this kind of issue.

**Comment by rhysd** [comment]
> I don't think that any [Default environment variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) are valid when referenced like this: ${{ env.GITHUB_WORKSPACE }}

Can you show any description which clearly states the `GITHUB_*` environment variables cannot be accessed with `${{ env.GITHUB_* }}`?

**Comment by georgespalding** [comment]
There should be some opportunity to improve validation since 
It is stated here that some vars cannot be overwritten. https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
![image](https://github.com/rhysd/actionlint/assets/478901/8ce83c17-6f31-480d-9e44-69ab168c9fdf)

This doc outlines the different contexts: https://docs.github.com/en/actions/learn-github-actions/contexts
![image](https://github.com/rhysd/actionlint/assets/478901/37bd24d5-5267-4e0e-b8ce-ec75fc1adf74)
The wording `variables set in a workflow, job, or step` tells me it does not include default environment variables.
This wording is repeated in https://docs.github.com/en/actions/learn-github-actions/contexts#env-context as well.
